js/autotrack/textInput.js

```diff
@@ -0,0 +1,37 @@
+import { getBaseComponentProps } from './common';
+import * as _ from 'lodash';
+
+export const autocaptureTextInputChange = track => (
+  eventType,
+  componentThis,
+  event
+) => {
+  // Attach a debounce function to the TextInput component instance if one's not already attached.
+  if (!componentThis.__heap__debounceTextChange) {
+    componentThis.__heap__debounceTextChange = _.debounce(
+      debouncedAutocaptureTextInputChange(track),
+      1000
+    );
+  }
+
+  componentThis.__heap__debounceTextChange(eventType, componentThis, event);
+};
+
+const debouncedAutocaptureTextInputChange = track => (
+  eventType,
+  componentThis,
+  event
+) => {
+  const autotrackProps = getBaseComponentProps(componentThis);
+
+  if (!autotrackProps) {
+    // We're not capturing this interaction.
+    return;
+  }
+
+  if (componentThis.props.placeholder) {
```

Comment:
Do we need to check that `componentThis` is real here?  Would it ever come through as null?  (might not be the worst thing to check regardless...)

Reply:
`componentThis` should never be `null`/`undefined`, but if it is, an error would happen in the `getBaseComponentProps()` call.  We should probably do some sort of invariant assertion within all instrumentation hook methods for these arguments

Reply:
👍